sstables/generation_type.hh

```diff
@@ -164,6 +200,44 @@ template <>
 struct fmt::formatter<sstables::generation_type> : fmt::formatter<std::string_view> {
     template <typename FormatContext>
     auto format(const sstables::generation_type& generation, FormatContext& ctx) const {
-        return fmt::format_to(ctx.out(), "{}", generation.as_int());
+        if (generation.is_uuid_based()) {
+            // format the uuid with 4 parts splitted with "_". each these parts is encoded
+            // as base36 chars.
+            //
+            // This matches the way how Cassandra formats UUIDBasedSSTableId, but we
+            // don't have to. just don't want to use "-" as the delimeter in UUID, as
+            // "-" is already used to split different parts in a SStable filename like
+            // "nb-1-big-Data.db".
+            const auto uuid = generation.as_uuid();
+            auto timestamp = ::utils::UUID_gen::decimicroseconds(uuid.timestamp());
+
+            char days_buf[4] = {};
+            auto days = std::chrono::duration_cast<std::chrono::days>(timestamp);
+            timestamp -= days;
+            char* days_end = std::to_chars(std::begin(days_buf), std::end(days_buf),
+                                           days.count(), 36).ptr;
+
+            char secs_buf[4] = {};
+            auto secs = std::chrono::duration_cast<std::chrono::seconds>(timestamp);
+            timestamp -= secs;
+            char* secs_end = std::to_chars(std::begin(secs_buf), std::end(secs_buf),
+                                           secs.count(), 36).ptr;
+
+            char decimicro_buf[5] = {};
+            char* decimicro_end = std::to_chars(std::begin(decimicro_buf), std::end(decimicro_buf),
+                                                timestamp.count(), 36).ptr;
+
+            char lsb_buf[13] = {};
+            char* lsb_end = std::to_chars(std::begin(lsb_buf), std::end(lsb_buf),
+                                          static_cast<uint64_t>(uuid.get_least_significant_bits()), 36).ptr;
+
+            return fmt::format_to(ctx.out(), "{:0>4}_{:0>4}_{:0>5}{:0>13}",
+                                  std::string_view(days_buf, days_end),
+                                  std::string_view(secs_buf, secs_end),
+                                  std::string_view(decimicro_buf, decimicro_end),
+                                  std::string_view(lsb_buf, lsb_end));
```

Comment:
Again, I think we should already have code which formats UUIDs, lets reuse it.

Reply:
It formats, but differently.

Reply:
i will revise the commit message and also add comment to explain why we have yet another parser and formatter for timeuuid.